modules/sdk-coin-atom/test/unit/atom.ts

```diff
@@ -336,4 +341,51 @@ describe('ATOM', function () {
       stub.restore();
     });
   });
+
+  describe('Recover transactions: ', () => {
+    const sandBox = sinon.createSandbox();
+    const destinationAddress = wrwUser.destinationAddress;
+    const coin = coins.get('tatom');
+    const testBalance = '150000';
+    const testAccoutNumber = '123';
+    const testChainId = 'test-chain';
+
+    beforeEach(() => {
+      const accountBalance = sandBox.stub(Atom.prototype, 'getAccountBalance' as keyof Atom);
+      accountBalance.withArgs(wrwUser.senderAddress).resolves(testBalance);
+
+      const accountNumber = sandBox.stub(Atom.prototype, 'getAccountNumber' as keyof Atom);
+      accountNumber.withArgs(wrwUser.senderAddress).resolves(testAccoutNumber);
+
+      const chainId = sandBox.stub(Atom.prototype, 'getChainId' as keyof Atom);
+      chainId.withArgs().resolves(testChainId);
+
+      const serializedEntChallenge = mockChallengeA;
+      const deserializedEntChallenge = Ecdsa.deserializeNtilde(serializedEntChallenge);
+      sinon.stub(rangeProof, 'generateNtilde').resolves(deserializedEntChallenge);
+    });
+
+    it('should recover a txn for non-bitgo recoveries', async function () {
+      const res = await basecoin.recover({
+        userKey: wrwUser.userKey,
+        backupKey: wrwUser.backupKey,
+        bitgoKey: wrwUser.bitgoKey,
+        walletPassphrase: wrwUser.walletPassphrase,
+        recoveryDestination: destinationAddress,
+      });
+      res.should.not.be.empty();
+      res.should.hasOwnProperty('serializedTx');
+      res.should.hasOwnProperty('scanIndex');
+      sandBox.assert.calledOnce(basecoin.getAccountBalance);
+      sandBox.assert.calledOnce(basecoin.getAccountNumber);
+      sandBox.assert.calledOnce(basecoin.getChainId);
```

Comment:
I think can you add a test case where you have n number of receive address with some funds and you get back an array of recovery transactions build for addresses which have funds